net/netcheck/netcheck.go

```diff
@@ -125,6 +129,44 @@ type Report struct {
 	// TODO: update Clone when adding new fields
 }
 
+// GetGlobalAddrs returns the v4 and v6 global addresses observed during the
+// netcheck, which includes the best latency endpoint first, followed by any
+// other endpoints that were observed repeatedly. It excludes singular endpoints
+// that are likely only the result of a hard NAT.
+func (r *Report) GetGlobalAddrs() ([]netip.AddrPort, []netip.AddrPort) {
```

Comment:
I'd name these result parameters, then reduce the duplicated types

   (v4s, v6s []netip.AddrPort)

Reply:
https://github.com/tailscale/tailscale/pull/12164